Userland/Libraries/LibWeb/Painting/PaintableBox.h

```diff
@@ -157,7 +157,7 @@ class PaintableBox : public Paintable {
     Optional<CSSPixelRect> mutable m_absolute_rect;
     Optional<CSSPixelRect> mutable m_absolute_paint_rect;
 
-    Optional<CSSPixelRect> mutable m_clip_rect;
+    Gfx::IntRect mutable m_clip_rect;
```

Comment:
probably it's not a good a idea to leak clip rectangle from painter. not sure if it would be better to save the whole painter state in PaintableBox and allow to restore Painter state not only from it's internal stack but from passed object. the problem I try to solve here is that painting of out-of-flow children should be continued using the clip rectangle of their containing block.

Reply:
Hmm, I think this feels less hackish than saving the entire `Gfx::Painter` state.